### ISSUE TITLE
handling ! in user path

### ISSUE THREAD

**Issue by androonuix** [NuixSDK/Communication-Scanner#3]
Hey Team,

My client requested guidance on finding communications between two and only two parties and this script is just the solution they need.  

They've encountered an error running the script that seems due to the fact that they have a '!' in their user path.  All client admin accounts are prefixed with the !.  there is no policy approved method of working around this with another non-admin account.

> (IOError) no !/ in spec 
> 
> (C:\Users\!clientuid\AppData\Roaming\Nuix\Scripts\CommunicationScanner.nuixscript\CommunicationScanner.rb:7) 

Is it possible to update the script to handle the "!" in the path?  

If not, might there be a shortcut/symlinky work around?

Appreciatively,
-Andrew

**Comment by JuicyDragon** [comment]
Hello Andrew,

I did some testing in 7.8.0.10 and it appears that in general, JRuby throws an error like this when attempting to `load`, `require` or `require_relative` dependencies and the dependency lives on a path containing a `!`.  I've tried escaping the `!` and URL encoding it (just in case) and neither seems to work.

Might they be able to instead keep the script in `C:\ProgramData\Nuix\Scripts`?  This would make it available to all users on the machine and should factor out the user name based path containing a `!`.

Beyond that they can modify their copy of the script to have hard coded non-relative paths.  They would probably need to at least update 2 lines that load up dependencies in the script.  If they define a suitable directory to hold these such as `C:\Temp\ScriptDependencies` (note there is no `!`) they would need to copy these 2 files to that directory:

- `Nx.jar`
- `AddressHelper.rb`

Then [line 7](https://github.com/Nuix/Communication-Scanner/blob/master/Ruby/CommunicationScanner.nuixscript/CommunicationScanner.rb#L7) of `CommunicationScanner.rb` needs to be modified from:

```ruby
require_relative "Nx.jar"
```

to an absolute path:

```ruby
require "C:\\Temp\\ScriptDependencies\\Nx.jar"
```

And then modify [line 21](https://github.com/Nuix/Communication-Scanner/blob/master/Ruby/CommunicationScanner.nuixscript/CommunicationScanner.rb#L21) of `CommunicationScanner.rb` from:

```ruby
load File.join(script_directory,"AddressHelper.rb")
```

to an absolute path:

```ruby
load "C:\\Temp\\ScriptDependencies\\AddressHelper.rb"
```

These changes tell the script to load those 2 files from the absolute path specified versus attempting to load them from a path relative to the main script file, which then allows them to host the dependencies on a path that does not contain the `!`.
